SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityTaskReconciliation.java

```diff
@@ -94,6 +95,17 @@ boolean isReconciliationRunning() {
   }
 
   public ReconciliationState startReconciliation() {
+    if (
+      (float) taskManager.getNumScheduledTasks() /
+      configuration.getOfferCacheSize() >=
```

Comment:
this was the closest value I could find for the denominator based on [these docs](http://getsingularity.com/Docs/reference/configuration.html#operational)

Reply:
Not quite the right thing here. We may actually need to instrument an executor to get this. Status updates are put in queue [here](https://github.com/HubSpot/Singularity/blob/master/SingularityService/src/main/java/com/hubspot/singularity/mesos/SingularityMesosSchedulerImpl.java#L742-L747) and put into the queue for [this executor service](https://github.com/HubSpot/Singularity/blob/master/SingularityService/src/main/java/com/hubspot/singularity/mesos/SingularityMesosStatusUpdateHandler.java#L647). Essentially what we want to inspect is the size of queued work on the executor service used in the status update handler

Reply:
Updated this to use the SingularityMesosStatusUpdateHandler: I added queue fullness function, but I wasn't sure if I should continue calculating fullness in SingularityTaskReconciliation with `configuration.getMesosConfiguration().getMaxStatusUpdateQueueSize()`